### DISCUSSION TITLE
How to add a form via GetForm.io

### DISCUSSION THREAD

**Discussion by imranakram** [chipzoller/hugo-clarity#223] [Q&A] [answered]
Hi, 
I'm a newbie in the Hugo world, I was wondering what would be the easiest way to add a GetForm.io based contact form on a website using this theme? Would be great if you could add some how-to instructions in the documentation. I've found an implementation in a ~~forked~~ theme [here](https://github.com/razonyang/hugo-theme-bootstrap/blob/master/layouts/_default/contact.html). There's a related config value there also. I have a very limited understanding of Hugo theme implementations in general so I don't know how easy it is to port this implementation here as a pull request perhaps. This would be a great feature in my opinion. What's your take on this @razonyang?

**Comment by razonyang** [comment, marked as answer]
## Layout

Create a layout called `layouts/_default/contact.html` with the following content:

```html
{{ define "main" }}
<form action="{{ .Site.Params.contact.endpoint }}" method="POST" enctype="multipart/form-data">
    <input type="text" name="name">
    <input type="email" name="email">
    <input type="file" name="file">
    <button type="submit">Send</button>
</form>
{{ end }}
```

## Page

Create a page called `content/contact.md` with the following content.

```markdown
+++
title = "Contact Us"
layout = "contact"
+++
```

> The `layout` must be same as the name we created above.

## Parameters

```toml
[contact]
  endpoint = "REPLACE_WITH_YOUR_GETFORM_ENDPOINT"
```

Now, you can access your contact page via /contact.

The page may look ugly, since I haven't used this awesome theme for a long time, so I can't help you with the CSS.

> BTW, the bootstrap theme is not a fork, it was created from scratch, so the styles, layouts and partials can't be compatibled with this theme.

**Reply by imranakram** [reply]
Thanks, I've corrected the "forked" part :)
